### PR TITLE
Gradle instructions

### PR DESCRIPTION
**Pull request by mernst**

### CODE REVIEW

File: docs/manual/checkerframework.gradle [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  }
+}
+
+// Checker Framework from Maven Central
+dependencies {
+  ext.checkerFrameworkVersion = '2.3.2'
+  checkerFrameworkAnnotatedJDK "org.checkerframework:jdk8:${checkerFrameworkVersion}"
+  checkerFrameworkJavac "org.checkerframework:compiler:2.3.2"
+  compileOnly "org.checkerframework:checker:${checkerFrameworkVersion}"
+  compile "org.checkerframework:checker-qual7:${checkerFrameworkVersion}"
```
Comment (smillst) [inline docs/manual/checkerframework.gradle]: This should be `checker-qual` unless using annotations in comments.

File: docs/manual/checkerframework.gradle [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+// build), use this instead:
+compileJava {
+  doFirst {
+    options.compilerArgs = [
+        // not done yet: org.checkerframework.checker.index.IndexChecker,
+        '-processor', 'org.checkerframework.checker.formatter.FormatterChecker,org.checkerframework.checker.lock.LockChecker,org.checkerframework.checker.nullness.NullnessChecker,org.checkerframework.checker.signature.SignatureChecker',
+        '-Xmaxerrs', '10000',
+        '-Xmaxwarns', '10000',
+        // '-Awarns',      // turn Checker Framework errors into warnings
+        '-source', '7', '-target', '7', // include if Java 7 compatibility is needed
```
Comment (smillst) [inline docs/manual/checkerframework.gradle]: This should probably be commented out.  We don't want to force Java 7.

File: docs/manual/checkerframework.gradle [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        '-AcheckPurityAnnotations',
+        "-Xbootclasspath/p:${configurations.checkerFrameworkAnnotatedJDK.asPath}",
+        "-Astubs=$System.env.CHECKERFRAMEWORK/checker/lib/javadoc.astub" // TODO: does not work when downloading from Maven Central
```
Comment (smillst) [inline docs/manual/checkerframework.gradle]: I don't think this is generally needed is it?

File: docs/manual/checkerframework-annos-in-comments.gradle [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+//   tasks.withType(JavaCompile).all { JavaCompile compile ->
+//     compile.doFirst {
+//       compile.options.compilerArgs = [
+//         '-processor', 'org.checkerframework.checker.formatter.FormatterChecker,org.checkerframework.checker.index.IndexChecker,org.checkerframework.checker.lock.LockChecker,org.checkerframework.checker.nullness.NullnessChecker,org.checkerframework.checker.signature.SignatureChecker',
+//         '-Xmaxerrs', '10000',
+//         '-Awarns',    // turn Checker Framework errors into warnings
+//         '-source', '7', '-target', '7', // include if Java 7 compatibility is needed
+//         '-AcheckPurityAnnotations',
+//         "-Xbootclasspath/p:${configurations.checkerFrameworkAnnotatedJDK.asPath}",
+//         "-Astubs=$System.env.CHECKERFRAMEWORK/checker/lib/javadoc.astub" // TODO: does not work when downloading from Maven Central
```
Comment (wmdietl) [inline docs/manual/checkerframework-annos-in-comments.gradle]: astub file was moved to `checker/resources/javadoc.astub`

File: docs/manual/checkerframework.gradle [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+// allprojects {
+//   tasks.withType(JavaCompile).all { JavaCompile compile ->
+//     compile.doFirst {
+//       compile.options.compilerArgs = [
+//         '-processor', 'org.checkerframework.checker.formatter.FormatterChecker,org.checkerframework.checker.index.IndexChecker,org.checkerframework.checker.lock.LockChecker,org.checkerframework.checker.nullness.NullnessChecker,org.checkerframework.checker.signature.SignatureChecker',
+//         '-Xmaxerrs', '10000',
+//         '-Awarns',    // turn Checker Framework errors into warnings
+//         '-AcheckPurityAnnotations',
+//         "-Xbootclasspath/p:${configurations.checkerFrameworkAnnotatedJDK.asPath}",
+//         "-Astubs=$System.env.CHECKERFRAMEWORK/checker/lib/javadoc.astub" // TODO: does not work when downloading from Maven Central
```
Comment (wmdietl) [inline docs/manual/checkerframework.gradle]: astub file was moved to `checker/resources/javadoc.astub`.

**Pull request merged**